third-party/qthread/qthread-src/src/threadqueues/mutexfifo_threadqueues.c

```diff
@@ -25,6 +25,17 @@
 #endif /* QTHREAD_USE_EUREKAS */
 #include "qt_subsystems.h"
 
+// Our onPark and onUnpark callbacks
+static void (*onPark)(void);
+static void (*onUnpark)(void);
+
+void qthread_registerOnPark(void (*_onPark)(void)) {
+    onPark = _onPark;
+}
+void qthread_registerOnUnpark(void (*_onUnpark)(void)) {
+    onUnpark = _onUnpark;
+}
+
```

Comment:
If I'm understanding correctly, some of these qthreads configurations have park/unpark operations shimmed out. Should qthread_registerOnPark halt or print out an error or something? I'm worried it will be a surprise to somebody in the future that it's not implemented somewhere.

Reply:
I've only implemented them in the core schedulers 'nemesis' and 'distrib' because the others are likely to be unused. I can add documentation in the qthread tasking layer documenting this, but the only real issue they would see is that they would leak memory when threads were idle.

Reply:
Done